spring-data-neo4j-rx/src/main/java/org/neo4j/springframework/data/core/Neo4jTemplate.java

```diff
@@ -319,16 +321,22 @@ private void processNestedAssociations(Neo4jPersistentEntity<?> neo4jPersistentE
 
 			Object value = propertyAccessor.getProperty(inverse);
 
-			Class<?> associationTargetType = inverse.getAssociationTargetType();
-
-			Neo4jPersistentEntity<?> targetNodeDescription = (Neo4jPersistentEntity<?>) neo4jMappingContext
-				.getPersistentEntity(associationTargetType);
-
 			Collection<RelationshipDescription> relationships = neo4jPersistentEntity.getRelationships();
 			RelationshipDescription relationship = relationships.stream()
 				.filter(r -> r.getFieldName().equals(inverse.getName()))
 				.findFirst().get();
 
+			Class<?> associationTargetType;
+			// if we have a relationship with properties, the targetNodeType is the Map Key
+			if (relationship.hasRelationshipProperties()) {
```

Comment:
Could be a conditional/ternary operation to align with the style below.

Reply:
Introduced a bit of refactoring with an own Context